lib/Db/LegacyRowMapper.php

```diff
@@ -149,7 +154,12 @@ private function resolveSearchValue(string $unresolvedSearchValue, string $userI
 		}
 	}
 
-	private function addOrderByRules(IQueryBuilder $qb, $sortArray) {
+	/**
+	 * @param (int|string)[][] $sortArray
+	 *
+	 * @psalm-param list<array{columnId: int, mode: 'ASC'|'DESC'}> $sortArray
```

Comment:
This doesn't seem related to the implementation but fine to annotate here, just the static analysis fails due to the incomplete array description as the array gets enriched with a columnType key

> Error: lib/Db/LegacyRowMapper.php:184:25: InvalidArrayOffset: Cannot access value on variable $sortRule using offset value of 'columnType', expecting 'columnId' or 'mode' (see https://psalm.dev/115)

Reply:
@juliushaertl Thank you for spot on! Wondering why the ci/cd checks keeps failing (Not immediately but after few days) 

Changed to this: @psalm-param list<array{columnId?: int, columnType?: string, mode?: 'ASC'|'DESC'}> $sortArray

Because when running psalm at my local there a case only 2 of them present (columnId & mode)

Reply:
The CI requires approval for outside contributors.

I'm also unsure why that may not happen for you locally, just checked and I get the same error. The only thing I could imagine is psalm cache which sometimes messes things up. You could try running with `composer run psalm -- --no-cache --threads=8` and see if that makes a difference.